### PR TITLE
ARROW-16198: [CI][Packaging][Python] Update VCPKG version

### PR DESCRIPTION
**Pull request by assignUser**

### CODE REVIEW

File: ci/vcpkg/ports.patch [hunk 1]
```diff
@@ -1,145 +1,3 @@
-diff --git a/ports/aws-c-auth/vcpkg.json b/ports/aws-c-auth/vcpkg.json
-index dc8f75e8..be703324 100644
---- a/ports/aws-c-auth/vcpkg.json
-+++ b/ports/aws-c-auth/vcpkg.json
-@@ -4,7 +4,7 @@
-   "port-version": 1,
-   "description": "C99 library implementation of AWS client-side authentication: standard credentials providers and signing.",
-   "homepage": "https://github.com/awslabs/aws-c-auth",
--  "supports": "!arm & !uwp",
```
Comment (assignUser) [inline ci/vcpkg/ports.patch:9]: They changed `!arm` to `!(windows & arm)` e.g. https://github.com/microsoft/vcpkg/blob/master/ports/aws-c-auth/vcpkg.json#L7
So I think these are no longer necessary?

File: ci/vcpkg/ports.patch [hunk 2]
```diff
@@ -174,23 +32,11 @@ index 2d6bba4d..0ac47887 100644
      PATCHES
          patch-relocatable-rpath.patch
          fix-aws-root.patch
+         fix-openssl3.patch
```
Comment (assignUser) [inline ci/vcpkg/ports.patch:35]: This line was added in the json, unsure if this patch is still necessary

**Comment by assignUser** [comment]
@kszucs could you take a look if this is correct?

**Comment by kszucs** [comment]
@github-actions crossbow submit *wheel*

**Comment by assignUser** [comment]
@kszucs I think most of those fails are caused by the vcpkg image not being up to date. I saw that the action in the corssbow repo for that runs on cron and can not be manually triggered for cases like this. I'll open a pr,

File: ci/vcpkg/ports.patch (hunk 1 above)
Comment (kszucs) [reply ci/vcpkg/ports.patch:9]: Then it indeed shouldn't be necessary.

**Comment by kszucs** [comment]
@assignUser that should only apply for the windows images, but the linux images get rebuilt in each run if they are not up to date.

**Comment by assignUser** [comment]
@github-actions crossbow submit python-wheel-manylinux-2014

**Comment by assignUser** [comment]
@github-actions crossbow submit _wheel_

**Comment by assignUser** [comment]
@github-actions crossbow submit -g wheel

**Comment by assignUser** [comment]
The `linux-headers` thing was a red hering the actual issue was: `Can't locate IPC/Cmd.pm in @INC (@INC contains:` 
Which seems to be fixed by installing `perl-IPC-Cmd` via yum. 
I'll push once the build finishes on my machine. 

@kszucs I don't really know why the macos build would fail due to permission issues (e.g. https://github.com/ursacomputing/crossbow/runs/6077317725?check_suite_focus=true#step:8:27 ) do you have any insights on what could cause that?

**Comment by assignUser** [comment]
@github-actions crossbow submit -g wheel

**Comment by kszucs** [comment]
> @kszucs I don't really know why the macos build would fail due to permission issues (e.g. https://github.com/ursacomputing/crossbow/runs/6077317725?check_suite_focus=true#step:8:27 ) do you have any insights on what could cause that?

The github actions service has stopped on our self-hosted machine and when I applied a workaround I started it with the wrong user. It should be fixed now.

**Comment by assignUser** [comment]
@github-actions crossbow submit -g wheel

File: ci/vcpkg/ports.patch (hunk 2 above)
Comment (assignUser) [reply ci/vcpkg/ports.patch]: This is patching the deprecated-declarations warning, does not work on manylinux2010 (gcc8) for some reason, overridden by -Wall maybe?

**Comment by kszucs** [comment]
@assignUser rebase on master in order to resolve the big-sur test errors

**Comment by assignUser** [comment]
@github-actions crossbow submit -g wheel

**Comment by assignUser** [comment]
Found the issue with the windows builds due to https://github.com/apache/arrow/pull/7388 cmake was looking for `zstd_static.lib` even though the version installed via vcpkg is called `zstd.lib` (since https://github.com/microsoft/vcpkg/commit/d8783052cbf85bb5b1777b58b998c88d99357943). I fixed this by manually setting the suffix to "", not sure if modifying the logic in `Findzstd.cmake` is necessary?

**Comment by assignUser** [comment]
I created an issue for adding the automatic change to the documentation: https://issues.apache.org/jira/browse/ARROW-16265

**Comment by kszucs** [comment]
> Found the issue with the windows builds, due to #7388 cmake was looking for `zstd_static.lib` even though the version installed via vcpkg is called `zstd.lib` (since [microsoft/vcpkg@d878305](https://github.com/microsoft/vcpkg/commit/d8783052cbf85bb5b1777b58b998c88d99357943)). I fixed this by manually setting the suffix to "", not sure if modifying the logic in `Findzstd.cmake` is necessary?

Nice catch! Could you set that cmake variable in `Usevcpkg.cmake`?

**Comment by kszucs** [comment]
I'm going to check locally what's the issue with absl in the universal2 builds.

**Comment by assignUser** [comment]
I created a job to dump the logs but that hasn't been run for over 2h ... Githubs Macs seem to be busy xD
https://github.com/assignUser/test-repo-a/actions/runs/2200637554

**Comment by assignUser** [comment]
I have finally (🤞) fixed the patches and am running a test. Were you able to find out what causes the issue with abseil @kszucs ? I think that should be the last piece to finish this up!

**Comment by kszucs** [comment]
Not yet, going to take a look.

**Comment by kszucs** [comment]
@assignUser can you trigger the windows image builds for the vcpkg version now that https://github.com/ursacomputing/crossbow/pull/14 has been merged?

**Comment by kszucs** [comment]
Than you can submit the windows wheel builds using `@github-actions crossbow submit wheel-windows-*` comment.

**Comment by assignUser** [comment]
@github-actions crossbow submit wheel-windows-*

**Comment by assignUser** [comment]
Both failures are caused by https://github.com/apache/arrow/pull/12930 rebasing should fix them.

**Comment by assignUser** [comment]
@github-actions crossbow submit wheel-windows-*

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp310-universal2

**Comment by kszucs** [comment]
https://github.com/apache/arrow/pull/12959 should resolve the universal2 issues

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

**Comment by kszucs** [comment]
@github-actions crossbow submit -g wheel

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-macos-big-sur-cp310-universal2

**Comment by kszucs** [comment]
`test-build-vcpkg-win` failure is unrelated to this PR

**Review by kszucs — APPROVED** [review]
Thanks @assignUser!

**Pull request closed without merging**